Lib/platform.py

```diff
@@ -614,17 +595,12 @@ def _syscmd_file(target, default=''):
         return default
     target = _follow_symlinks(target)
     try:
-        proc = subprocess.Popen(['file', target],
-                stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-
-    except (AttributeError, OSError):
+        output = subprocess.check_output(['file', target],
+                                         stderr=subprocess.DEVNULL,
+                                         encoding='latin-1')
+    except (AttributeError, OSError, subprocess.CalledProcessError):
```

Comment:
Why an AttributeError is caught here?

Reply:
AttributeError is to catch... missing os.popen() function (I guess) :-) AttributeError is as old as the module itself: commit 246d847475d041a823dbc3cc3aceae46e97de937. Do you think that we should remove it?

Reply:
Guys, please remember that these modules are used by other Python implementations as well. The AttributeError most likely came from this: having e.g. IronPython or Jython not having the needed os API available.

Reply:
Other Python implementations are supposed the implement the same stdlib. Why would IronPython or Jython not have subprocess.check_output()? For example, if IronPython implements "Python 3.5", they should copy platform.py from CPython 3.5.

If they decide to not implement subprocess.check_output(), well, we can have a talk :-) In the meanwhile, PyPy does patch the stdlib, but they don't propose these patches upstream :-(

Reply:
This used to be different... could be that we didn't listen to them.

In any case, see here for when the AttributeErrors were added: https://github.com/python/cpython/blob/2b665dbb53d02be09947116c1fe2ac002375d348/Lib/platform.py#L64

In Python 2 the module was meant to work across many different Python versions and platforms. It would be good to keep this feature in place, since it needs to be updated more frequently than the whole Python version stack (i.e. applications should be able to copy a more recent from from 3.7 to 3.5 and still have it work).

Reply:
Well, subprocess and and os.popen() are now available available on all platforms. The situation became better in Python 3 ;-)

Reply:
Note: PyPy3 has os.popen() and subprocess, and Jython and IronPython are both stuck at Python 2.7 ;-)

Reply:
`os.popen()` is optional in Python 2, but always available in Python 3. If other implementation have enough different stdlib without `os.popen()` and `subprocess.check_output()`, it can have different `platform` module as well.

I believe it is safe to remove AttributeError in all Python 3 versions.